src/lib.rs

```diff
@@ -202,9 +212,24 @@ impl Decode for Method {
             ]));
         }
 
+        let mut constraints: Option<Vec<u8>> = None;
+        let constraints_count = block_cooldown_and_constraints_count >> 1;
+        if constraints_count > 0 {
+            let mut constraints_buf: Vec<u8> = Default::default();
+            for _ in 0..constraints_count {
+                constraints_buf.push(input.read_byte()?);
+            }
+            let _ = match Contract::decode(&constraints_buf) {
```

Comment:
nitpick: an `if let Err(_) = Contract::decode(&constraints_buf)` would be nicer here